### PR TITLE
13 lang selection page

### PR DESCRIPTION
**Pull request by Digvorzhak**
Hello,
This pull request is my biggest so far and covers the entire page for language selection. 
Tested, and so far works good for me. 
New implementations: 
- Updated Header and HeaderLayout (VERY IMPORTANT SINCE THE PREVIOUS ONE WASN'T DYNAMIC). Now each page will render Header first and will be able to send props to cover all  use cases.
- Updated react router to work properly with HeaderLayout. (only pages that contain green header will be placed as children for HeaderLayout component).
- index.jsx files for each folder. I have imported and exported every file from index.jsx only. This way it's much easier to read the code for each page. 
- New logic to styles from branch 12, making buttons more dynamic and easy to figure out what's is going on. (for example, if  the button is inactive it will grayed out. will change to green when it's allowed for user of course).

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -1,10 +1,21 @@
 import Intro from "./pages/Intro";
+import LangSelection from "./pages/LangSelection";
+import Interest from "./pages/Interest";
 import { createBrowserRouter, RouterProvider } from "react-router-dom";
-import Occupation from "./pages/Occupation";
```
Comment (yishain11) [inline client/src/App.jsx:3]: don't remove, this is not your code, unsless you KNOW that its not used anymore

File: client/src/styles/StyledHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  font-weight: 500;
+  font-size: 1.125rem;
+  color: #FFFFFF;
+  position: absolute;
+  @media (min-width: 700px) {
+    padding: 3rem;
+    height: 12.5rem;
+  }
+`;
+
```
Comment (yishain11) [inline client/src/styles/StyledHeader.jsx:20]: remove empty lines

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/src/App.jsx (hunk 1 above)
Comment (Digvorzhak) [reply client/src/App.jsx:3]: understood, I reverted this line of code

File: client/src/styles/StyledHeader.jsx (hunk 2 above)
Comment (Digvorzhak) [reply client/src/styles/StyledHeader.jsx]: done

File: client/src/styles/StyledLanguageButton.jsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+export const StyledLanguageButton = styled.button`
+  width: 17rem;
+  height: 4.4rem;
+  border: 0.1rem solid #50924E;
+  border-radius: 0.8rem;
+  padding: 1.2rem 1rem;
+  cursor: pointer;
+  background-color: ${({bg}) => bg || "#FFFFFF"};
+  font-family: "Mulish", sans-serif;
```
[the remark this answers is not in the document]
Comment (Digvorzhak) [reply client/src/styles/StyledLanguageButton.jsx:11]: for "#FFFFFF"? without qutoes it doesn't work

File: client/src/pages/Intro.jsx [hunk 4]
```diff
@@ -1,36 +1,24 @@
...
+      <StyledMargin direction='vertical' margin='2rem'/>
+      <StyledSpan>Hi!</StyledSpan>
+      <StyledDiv>
+      <StyledParagraph>
+      Please answer some quick
+      question so we can find you
+      relevant people to chat with 
+      </StyledParagraph>
+      </StyledDiv>
+      <StyledButton to="/lang" children={"Lets Do It"}/>
```
Comment (yishain11) [inline client/src/pages/Intro.jsx:20]: don't use children as a prop name. children means what is between the tags. call it "text" or something like that

File: client/src/pages/LangSelection.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            <StyledLanguageButton bg={language.value === "HE" ? "#50924E" : "#FFFFFF"} color={language.value === "HE" ? "#FFFFFF" : "#000000"} onClick={() => setLanguage({...language, value: "HE"})}>Hebrew</StyledLanguageButton>
+            </Flex>
+            <StyledButton to={language.value ? "/interests" : null} onClick={() => {
+              if(!language.value){
+                return
+              }
+              dispatch(addUserDataField(language))
+              setLanguage({...language, value: ""})
+            }
+            } bg={language.value ? "#50924E" : "#d7ddd6"} hoverBg={language.value ? "#396d37" : "#d7ddd6"} children={"Save & Next"}></StyledButton>
```
Comment (yishain11) [inline client/src/pages/LangSelection.jsx:45]: don't use children as a prop name. children means what is between the tags. call it "text" or something like that

**Review by yishain11 — CHANGES_REQUESTED** [review]

**Pull request merged**